### PR TITLE
Added solution to the task.

### PR DESCRIPTION
**Pull request by yanna-stepanova**

### CODE REVIEW

File: src/main/java/core/basesyntax/Application.java [hunk 1]
```diff
@@ -3,5 +3,10 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Ball[] balls = new Ball[3];
+        for (Ball ball : balls) {
+            ball = new Lottery().getRandomBall();
+            System.out.println(ball.toString());
+        }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:10]: toString is called under the hood automatically
try to rewrite this without using array and creating only one lottery instance, not 3

File: src/main/java/core/basesyntax/Ball.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    public Ball(String color, int number) {
+        this.color = color;
+        this.number = number;
+    }
+
+    @Override
+    public String toString() {
+        return "The ball has " + this.color + " color and number " + this.number;
+    }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Ball.java:15]: we use `this` when we need to differentiate between a class field and an argument that share the same name, it's not the case here, so you can safely omit them

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        int index = new Random().nextInt(Color.values().length);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: make random a private class-level field

File: src/main/java/core/basesyntax/Lottery.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        return new Ball(new ColorSupplier().getRandomColor(), new Random().nextInt(100));
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:7]: make random a private class-level field
make color supplier a private class-level field
make 100 a private class-level constant

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job overall!

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 5]
```diff
@@ -1,7 +1,16 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
+    private Random random = new Random();
+
+    public Random getRandom() {
+        return random;
+    }
```
Comment (olekskov) [inline src/main/java/core/basesyntax/ColorSupplier.java:10]:
```suggestion
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+public class Lottery {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private final int size = 100;
+
+    public Ball getRandomBall() {
+        return new Ball(colorSupplier.getRandomColor(), colorSupplier.getRandom().nextInt(size));
+    }
```
Comment (olekskov) [inline src/main/java/core/basesyntax/Lottery.java:9]:
```suggestion
public class Lottery {
    private static final int MAX_NUMBER = 100;

    private Random random = new Random();
    private ColorSupplier colorSupplier = new ColorSupplier();
    
    public Ball getRandomBall() {
        return new Ball(colorSupplier.getRandomColor(), random.nextInt(MAX_NUMBER));
    }
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 7]
```diff
@@ -1,7 +1,16 @@
...
+
     public String getRandomColor() {
-        return null;
+        int index = getRandom().nextInt(Color.values().length);
```
Comment (olekskov) [inline src/main/java/core/basesyntax/ColorSupplier.java:13]:
```suggestion
        int index = random.nextInt(Color.values().length);
```

**Review by olekskov — CHANGES_REQUESTED** [review]
Oleh's comment was a bit misunderstood, please check comments to fix

File: src/main/java/core/basesyntax/Application.java [hunk 8]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+        Ball[] balls = {lottery.getRandomBall(), lottery.getRandomBall(), lottery.getRandomBall()};
+        System.out.println(Arrays.toString(balls));
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:10]: it's an interesting approach, but let's use a for loop for this, as mentioned in the [checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery)
image

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by okuzan — CHANGES_REQUESTED** [review]
review for sync with the platform

File: src/main/java/core/basesyntax/Application.java [hunk 9]
```diff
@@ -1,7 +1,15 @@
...
+
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+        Ball[] balls = new Ball[3];
+        for (int i = 0; i < balls.length; i++) {
+            balls[i] = lottery.getRandomBall();
+        }
+        System.out.println(Arrays.toString(balls));
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:13]:
```suggestion
        for (int i = 0; i < BOUND; i++) {
             System.println(lottery.getRandomBall());
        }
```

**Review by okuzan — CHANGES_REQUESTED** [review]
let's print balls directly in the loop

**Comment by yanna-stepanova** [comment]
Should  the lottery instance be as a private class-level field in Application.java?

File: src/main/java/core/basesyntax/Application.java [hunk 10]
```diff
@@ -1,7 +1,13 @@
 package core.basesyntax;
 
 public class Application {
+    private static final int BOUND = 3;
+    private static Lottery lottery = new Lottery();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:5]: not necessary, but ok

**Review by okuzan — APPROVED** [review]
Well done! Best of luck on your educational journey 🤞 🚀

**Pull request closed without merging**